models/refugee.py

```diff
@@ -0,0 +1,105 @@
+from datetime import date, datetime
+
+class Refugee:
+    """
+    A class to represent a refugee family.
+    """
+    def __init__(self,
+                 name: str,
+                 camp: str,
+                 medicalCondition: str,
+                 numOfFamilyMember: int,
+                 dateOfClosing=None):
+
+        """
+        Initialise a new refugee.
+        :param name of refugee family
+        :param number of family member
+        :param camp of the refugee family locating at
+        :param medical condition of the refugee
+        :param start date of refugee family
+        :param closing date of refugee family
+        """
+
+        self.name = name
+        self.numOfFamilyMember = self.__checkNumOfFamilyMember(numOfFamilyMember)
+        self.camp = camp
+        self.medicalCondition = medicalCondition
+        self.dateOfCreation = datetime.today().date()
+        self.dateOfClosing = self.__checkDateOfClosing(dateOfClosing)
+
+    def formatName(self, name):
+        """
+        format name to a new name
+        """
+        self.name = name
+        #error handling if invalid name is inputed
+
+    def __checkNumOfFamilyMember(self, numOfFamilyMember):
+        """
+        format number of family member
+        """
+        if not isinstance(numOfFamilyMember, int) or numOfFamilyMember < 0:
+            raise self.InvalidNumOfFamilyMemberException()
+        return numOfFamilyMember + 1
+
+    def formatCamp(self, newCamp):
+        """
+        format identification camp of refugee
+        """
+        if newCamp not in camp.Camp.camps:
+            raise self.InvalidCampException(newCamp)
+        self.camp = newCamp
+
+    def formatMedicalCondition(self, medicalCondition): #medical condition types provided or a description?
+        """
+        format refugee family medical condition
+        """
+        self.medicalCondition = medicalCondition
+
+    def __checkDateOfClosing(self, dateOfClosing: date): #for removing and archive
+        """
+        format closing date of refugee family
+        """
+        today = datetime.today().date()
+        if today >= dateOfClosing:
+            raise self.InvalidClosingDateException()
+        elif dateOfClosing == None:
+            return dateOfClosing
+        else:
+            return dateOfClosing
```

Comment:
Does this mean that the refugee's profiles are still kept after they left the camps?
What is the motivation for keeping them?
I am not sure if we need this, and it may complicate other operations on refugees.

Reply:
Sorry, just noticed that we do need to archive refugees. 
If that is the case, we might need some easy way to check if this refugee is still active